src/core/server/http/http_config.ts

```diff
@@ -31,102 +32,108 @@ const match = (regex: RegExp, errorMsg: string) => (str: string) =>
 
 // before update to make sure it's in sync with validation rules in Legacy
 // https://github.com/elastic/kibana/blob/master/src/legacy/server/config/schema.js
-export const config = {
-  path: 'server',
-  schema: schema.object(
-    {
-      name: schema.string({ defaultValue: () => hostname() }),
-      autoListen: schema.boolean({ defaultValue: true }),
-      basePath: schema.maybe(
-        schema.string({
-          validate: match(validBasePathRegex, "must start with a slash, don't end with one"),
-        })
-      ),
-      cors: schema.conditional(
-        schema.contextRef('dev'),
-        true,
-        schema.object(
-          {
-            origin: schema.arrayOf(schema.string()),
+const configSchema = schema.object(
+  {
+    name: schema.string({ defaultValue: () => hostname() }),
+    autoListen: schema.boolean({ defaultValue: true }),
+    basePath: schema.maybe(
+      schema.string({
+        validate: match(validBasePathRegex, "must start with a slash, don't end with one"),
+      })
+    ),
+    cors: schema.conditional(
+      schema.contextRef('dev'),
+      true,
+      schema.object(
+        {
+          origin: schema.arrayOf(schema.string()),
+        },
+        {
+          defaultValue: {
+            origin: ['*://localhost:9876'], // karma test server
           },
-          {
-            defaultValue: {
-              origin: ['*://localhost:9876'], // karma test server
-            },
-          }
-        ),
-        schema.boolean({ defaultValue: false })
+        }
       ),
-      customResponseHeaders: schema.recordOf(schema.string(), schema.any(), {
-        defaultValue: {},
-      }),
-      host: schema.string({
-        defaultValue: 'localhost',
-        hostname: true,
-      }),
-      maxPayload: schema.byteSize({
-        defaultValue: '1048576b',
-      }),
-      port: schema.number({
-        defaultValue: 5601,
-      }),
-      rewriteBasePath: schema.boolean({ defaultValue: false }),
-      ssl: sslSchema,
-      keepaliveTimeout: schema.number({
-        defaultValue: 120000,
-      }),
-      socketTimeout: schema.number({
-        defaultValue: 120000,
-      }),
-      compression: schema.object({
-        enabled: schema.boolean({ defaultValue: true }),
-        referrerWhitelist: schema.maybe(
```

Comment:
The diff is a mess because I had to extract the schema to its own var to add deprecations, but the only real change within the schema is here.